### PR TITLE
Document Layout Instability API

### PR DESCRIPTION
**Pull request by Elchi3**
### Description

This PR aims to complete the reference docs for the https://wicg.github.io/layout-instability/ API. It also integrates it into the rest of the Performance API docs per our plan to have all Performance APIs in one group and at a central place.

### Motivation

https://github.com/openwebdocs/project/issues/62

### Additional details

I was thinking about a checklist for what to do when documenting a new Performance API. I haven't come too far on that. 
- Update the main Performance_API page
- Update the diagram
- Mention the new API in the PerformanceEntry interface and update its sub pages.
- Update GroupData to add new interfaces to the "Performance" group
- Add new reference pages with the sidebar `{{APIRef("Performance API")}}`
- Add a chapter to the guide (TBD)
- Add new glossary pages for new metric name (if any)
- Update self.performance page to talk about worker availability.

### Related issues and pull requests

None.

### CODE REVIEW

**Comment by wbamberg** [comment]
I'll look at this one on Monday, Florian.

File: files/en-us/web/api/layoutshift/index.md [hunk 1]
```diff
@@ -14,22 +14,41 @@ tags:
...
 ## Instance properties
 
+This interface extends the following {{domxref("PerformanceEntry")}} properties for `layout-shift` performance entry types by qualifying them as follows:
+
+- {{domxref("PerformanceEntry.duration")}} {{ReadOnlyInline}} {{Experimental_Inline}}
+  - : Always returns `0` (the concept of duration does not apply to layout shifts).
+- {{domxref("PerformanceEntry.entryType")}} {{ReadOnlyInline}} {{Experimental_Inline}}
+  - : Always returns `"layout-shift"`
+- {{domxref("PerformanceEntry.name")}} {{ReadOnlyInline}} {{Experimental_Inline}}
+  - : Always returns `"layout-shift"`
```
[the remark this answers is not in the document]
Comment (wbamberg) [reply files/en-us/web/api/layoutshift/index.md:38]: This seems to be `undefined` in Chrome (but that seems wrong according to the spec).

File: files/en-us/web/api/layoutshift/value/index.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+slug: Web/API/LayoutShift/value
+page-type: web-api-instance-property
+browser-compat: api.LayoutShift.value
+tags:
+  - experimental
+---
+
+{{SeeCompatTable}}{{APIRef("Performance API")}}
+
+The **`value`** readonly property of the {{domxref("LayoutShift")}} interface returns the layout shift score calculated by the impact fraction (fraction of the viewport that was shifted) times the distance fraction (distance moved as a fraction of viewport).
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshift/value/index.md:12]:
```suggestion
The **`value`** read-only property of the {{domxref("LayoutShift")}} interface returns the layout shift score calculated as the impact fraction (fraction of the viewport that was shifted) multiplied by the distance fraction (distance moved as a fraction of viewport).
```

File: files/en-us/web/api/layoutshift/value/index.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Value
+
+A number between `0.0` and `1.0` indicating the layout shift score.
+
+It is calculated by the impact fraction (fraction of the viewport that was shifted) times the distance fraction (distance moved as a fraction of viewport).
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshift/value/index.md:18]:
```suggestion
It is calculated as the impact fraction (fraction of the viewport that was shifted) multiplied by the distance fraction (distance moved as a fraction of viewport).
```

File: files/en-us/web/api/layoutshift/value/index.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+```plain
+layout shift score = impact fraction * distance fraction
+```
+
+For more details, see [Layout shift score](https://web.dev/cls/#layout-shift-score) on web.dev.
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshift/value/index.md:24]: It's a bit sad to defer to web.dev for this but I see where you are coming from 🤷‍♂️

File: files/en-us/web/api/layoutshiftattribution/index.md [hunk 5]
```diff
@@ -41,7 +41,9 @@ The following example finds the element that is causing the largest layout shift
 
 ```js
 function getCLSDebugTarget(entries) {
-  const largestEntry = entries.reduce((a, b) => a && a.value > b.value ? a : b);
+  const largestEntry = entries.reduce((a, b) =>
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshiftattribution/index.md:44]: I think the description at line 40 is not quite right. It doesn't "finds the element that is causing the largest layout shift, and prints that `node` to the console". AFAICT, it finds the entry with the highest layout shift score, then logs the element in that entry with the largest size prior to the shift (`previousRect`).

File: files/en-us/web/api/layoutshiftattribution/index.md [hunk 6]
```diff
@@ -15,9 +15,9 @@ tags:
 browser-compat: api.LayoutShiftAttribution
 ---
 
-{{APIRef("Layout Instability API")}}{{SeeCompatTable}}
+{{APIRef("Performance API")}}{{SeeCompatTable}}
 
-The `LayoutShiftAttribution` interface of the [Layout Instability API](/en-US/docs/Web/API/Layout_Instability_API) provides debugging information about elements which have shifted.
+The `LayoutShiftAttribution` interface provides debugging information about elements which have shifted.
 
 Instances of `LayoutShiftAttribution` are returned in an array by calling {{domxref("LayoutShift.sources")}}.
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshiftattribution/index.md:22]: Line 26 `Node` -> `node`

File: files/en-us/web/api/layoutshiftattribution/node/index.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ tags:
 browser-compat: api.LayoutShiftAttribution.node
 ---
 
-{{APIRef("Layout Instability API")}}{{SeeCompatTable}}
+{{APIRef("Performance API")}}{{SeeCompatTable}}
 
 The **`node`** read-only property of the {{domxref("LayoutShiftAttribution")}} interface returns a {{domxref("node")}} representing the object that has shifted.
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshiftattribution/node/index.md:18]: Line 21 `node`-> `Node`.

File: files/en-us/web/api/layoutshiftattribution/node/index.md [hunk 8]
```diff
@@ -12,7 +12,7 @@ tags:
...
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshiftattribution/node/index.md:17]:
```suggestion
The **`node`** read-only property of the {{domxref("LayoutShiftAttribution")}} interface returns a {{domxref("Node")}} representing the object that has shifted.
```

**Review by wbamberg — CHANGES_REQUESTED** [review]

File: files/en-us/glossary/cumulative_layout_shift/index.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+---
+title: Cumulative Layout Shift (CLS)
+slug: Glossary/Cumulative_Layout_Shift
+---
+
+**Cumulative Layout Shift** (CLS) is a score that is the sum of every layout shift value is recorded. Layout shifts happen when any element that is visible in the viewport changes its position between two frames.
+
+It is measured using the {{domxref("LayoutShift")}} API and a {{domxref("PerformanceObserver")}} by summing the values as they reported to the observer. The final score is taken at the time of the [`visibilitychange` event](/en-US/docs/Web/API/Document/visibilitychange_event).
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply files/en-us/glossary/cumulative_layout_shift/index.md:8]: I had a look at the web.dev article but I mostly referred to non-normative text from the spec: https://wicg.github.io/layout-instability/#cumulative-layout-shift-cls-score

> The cumulative layout shift score is expected to have a general correspondence to the severity of layout instability for the lifetime of a page.

> The developer can use this API to compute the DCLS or CLS scores, by summing the values as they are reported to the observer, and taking the "final" score at the time of the [visibilitychange event](https://w3c.github.io/page-visibility/#dom-document-onvisibilitychange).

Reading the web.dev article again, there is this:
> Previously CLS measured the sum total of all individual layout shift scores that occurred during the entire lifespan of the page. To see which tools still provide the ability to benchmark against the original implementation, check out [Evolving Cumulative Layout Shift in web tooling](https://web.dev/cls-web-tooling).

I guess CLS is a concept built on top of this API and basically anyone is free to implement whatever scores on top of it. I guess we could change it to the new Google-defined score? https://web.dev/cls/#measure-cls-in-javascript? Or we could present both and say that you can implement arbitrary scores here?

About that new code, though:

> This code shows the basic way to calculate and log CLS. However, accurately measuring CLS in a way that matches what is measured in the [Chrome User Experience Report](https://developer.chrome.com/docs/crux/) (CrUX) is more complicated

Then Web Vitals is mentioned as a library for a more precise Google-defined CLS score.

File: files/en-us/web/api/layoutshift/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Elchi3) [reply files/en-us/web/api/layoutshift/index.md:38]: Yeah I filed https://github.com/WICG/layout-instability/issues/117

File: files/en-us/web/api/layoutshift/value/index.md (hunk 4 above)
Comment (Elchi3) [reply files/en-us/web/api/layoutshift/value/index.md:24]: Yeah I think there is a lot to unpack for the calculation of `value`. I wasn't sure how and if to go into this. The web.dev page already did this work, so I thought we might as well reference it. Referencing the spec leads to a much more cryptic explanation https://wicg.github.io/layout-instability/#sec-layout-shift-value :(

File: files/en-us/glossary/cumulative_layout_shift/index.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (Elchi3) [reply files/en-us/glossary/cumulative_layout_shift/index.md]: I removed the glossary and referred to https://web.dev/cls/ as an example how people use this API

**Comment by Elchi3** [comment]
Sorry for being so slow to get back into this. I removed CLS for now and presented it more like a metric that people derive with the help of this API but it might not be the only one. The examples also no longer attempt to do (outdated) CLS calculations.

**Comment by Elchi3** [comment]
This is ready for a re-review, fwiw. I keep getting merge conflicts 😅

File: files/en-us/web/api/layoutshift/hadrecentinput/index.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+The following example shows how the `hadRecentInput` property is used to only count layout shifts without recent user input.
+
+```js
+const observer = new PerformanceObserver((list) => {
+  for (const entry of list.getEntries()) {
+    // Count layout shifts without recent user input only
+    if (!entry.hadRecentInput) {
+      console.log("LayoutShift value:", entry.value);
+      if (entry.sources) {
+        for (const { node, curRect, prevRect } of entry.sources)
```
Comment (wbamberg) [inline files/en-us/web/api/layoutshift/hadrecentinput/index.md:33]: Throughout all these examples, `curRect` and `prevRect` need to be `currentRect` and `previousRect`.

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks Florian!

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks Florian!

**Review by wbamberg — APPROVED** [review]
👍 thank you @Elchi3 !

**Pull request merged**